paddlenlp/taskflow/information_extraction.py

```diff
@@ -114,6 +115,353 @@ def get_dynamic_max_length(examples, default_max_length: int, dynamic_max_length
+                    eos_token_id=self._tokenizer.eos_token_id,
+                    pad_token_id=self._tokenizer.pad_token_id,
+                    num_return_sequences=self._num_return_sequences,
+                    use_cache=True,
+                )
+                # result = result[0]
+                results.extend(result)
+        out_list = []
+        for x in results:
+            if self._static_mode:
+                res = self._tokenizer.decode(x, skip_special_tokens=True)
+                res = res.strip("\n")
+            else:
+                res = self._tokenizer.decode(x.numpy().tolist(), skip_special_tokens=True)
+                res = res.strip("\n")
+            out_list.append([{"text": res}])
+        
+        print('out_List:',out_list)
+        return out_list
+
+    def _multi_stage_predict(self,data):
+        """
+        Traversal the schema tree and do multi-stage prediction.
+
+        Args:
+            data (list): a list of strings
+
+        Returns:
+            list: a list of predictions, where the list's length
+                equals to the length of `data`
+        """
+        results = [{} for _ in range(len(data))]
+        print(results)
+        # Input check to early return
+        if len(data) < 1 or self._schema_tree is None:
+            return results
+
+        # Copy to stay `self._schema_tree` unchanged
+        schema_list = self._schema_tree.children[:]
+        print("schema_list:",schema_list)
+        while len(schema_list) > 0:
+            node = schema_list.pop(0)
+            examples = []
+            input_map = {}
+            cnt = 0
+            idx = 0
+            if not node.prefix:
+                for one_data in data:
+                    examples.append(
+                        {
+                            "text": one_data,
+                            "prompt": dbc2sbc(node.name),
+                        }
+                    )
+                    input_map[cnt] = [idx]
+                    idx += 1
+                    cnt += 1
+            else:
+                for pre, one_data in zip(node.prefix, data):
+                    if len(pre) == 0:
+                        input_map[cnt] = []
+                    else:
+                        for p in pre:
+                            if self._is_en:
+                                if re.search(r"\[.*?\]$", node.name):
+                                    prompt_prefix = node.name[: node.name.find("[", 1)].strip()
+                                    cls_options = re.search(r"\[.*?\]$", node.name).group()
+                                    # Sentiment classification of xxx [positive, negative]
+                                    prompt = prompt_prefix + p + " " + cls_options
+                                else:
+                                    prompt = node.name + p
+                            else:
+                                prompt = p + node.name
```

Comment:
```suggestion
                            prompt = p + node.name
```
这段特殊处理的分支，可以不要吧。 cls_options 的操作应该没有啥用吧

Reply:
对